subsys/fs/fs_impl.h

```diff
@@ -28,6 +29,38 @@ extern "C" {
 const char *fs_impl_strip_prefix(const char *path,
 				 const struct fs_mount_t *mp);
 
+/**
+ * @brief Searches for file system type FS_* identifier, as defined in fs/fs.h,
+ *        compatible with string representation of type.
+ *
+ * @param string representation of type.
+ *
+ * @return fs type or FS_UNKNOWN.
+ */
+int fs_get_compatible(const char *type_sz);
```

Comment:
Maybe `fs_lookup_by_compatible()`?  

Also, is anybody going to do anything with this index value other than get the corresponding API structure?  If not, the function can return the API structure or `NULL`.